### PR TITLE
[Azure.Monitor.Query] fix pipeline and update readme

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ For Metrics queries across multiple Azure resources, use the following client:
 
 ```C# Snippet:CreateMetricsClient
 var client = new MetricsClient(
-    new Uri("https://metrics.monitor.azure.com/.default"),
+    new Uri("https://<region>.metrics.monitor.azure.com/"),
```
Comment (pallavit) [inline sdk/monitor/Azure.Monitor.Query/README.md:59]: Do we not need default anymore?

**Review by pallavit — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.Query/README.md (hunk 1 above)
Comment (nisha-bhatia) [reply sdk/monitor/Azure.Monitor.Query/README.md:59]: The .default is needed for the scope so adding it here made the snippet incorrect :)

File: sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs [hunk 2]
```diff
@@ -24,11 +24,11 @@ public class MetricsClient
         /// Creates an instance of <see cref="MetricsClient"/>.
         /// <code snippet="Snippet:CreateMetricsClient" language="csharp">
         /// var client = new MetricsClient(
-        ///     new Uri(&quot;https://metrics.monitor.azure.com/.default&quot;),
+        ///     new Uri(&quot;https://&lt;region&gt;.metrics.monitor.azure.com/&quot;),
         ///     new DefaultAzureCredential());
         /// </code>
         /// </summary>
-        /// <param name="endpoint">The data plane service endpoint to use. For example, <c>https://metrics.monitor.azure.com/.default</c> for Azure Public Cloud.</param>
+        /// <param name="endpoint">The data plane service endpoint to use. For example, <c>https://(region).metrics.monitor.azure.com/</c> for Azure Public Cloud.</param>
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/src/MetricsClient.cs:31]: For consistency in placeholder syntax, use less than/greater than symbols
```suggestion
        /// <param name="endpoint">The data plane service endpoint to use. For example, <c>https://&lt;region&gt;.metrics.monitor.azure.com/</c> for Azure Public Cloud.</param>
```

File: sdk/monitor/Azure.Monitor.Query/tests/MetricsQueryClientSamples.cs [hunk 3]
```diff
@@ -223,9 +223,15 @@ public async Task QueryResourcesMetricsWithOptions()
             string resourceId = TestEnvironment.StorageAccountId;
 #endif
             #region Snippet:CreateMetricsClient
+#if SNIPPET
             var client = new MetricsClient(
-                new Uri("https://metrics.monitor.azure.com/.default"),
+                new Uri("https://<region>.metrics.monitor.azure.com/"),
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/tests/MetricsQueryClientSamples.cs:228]: Is the trailing slash necessary?

Comment (nisha-bhatia) [reply sdk/monitor/Azure.Monitor.Query/tests/MetricsQueryClientSamples.cs]: no that is a typo - thank you

File: sdk/monitor/Azure.Monitor.Query/README.md [hunk 4]
```diff
@@ -679,12 +679,13 @@ The `QueryResources` method also accepts a `MetricsQueryResourcesOptions`-typed
...
     "/subscriptions/<id>/resourceGroups/<rg-name>/providers/<source>/storageAccounts/<resource-name-1>";
 var client = new MetricsClient(
-    new Uri("https://metrics.monitor.azure.com/.default"),
+    new Uri("https://<region>.metrics.monitor.azure.com"),
     new DefaultAzureCredential());
 var options = new MetricsQueryResourcesOptions
 {
     OrderBy = "sum asc",
-    RollUpBy = { "RollUpBy=City" }
+    Size = 10,
```
Comment (scottaddie) [inline sdk/monitor/Azure.Monitor.Query/README.md:687]: Please update the text immediately before this code sample. Note that it currently mentions `RollUpBy`.

File: sdk/monitor/Azure.Monitor.Query/tests/MonitorQueryTestEnvironment.cs [hunk 5]
```diff
@@ -28,7 +28,6 @@ public class MonitorQueryTestEnvironment : TestEnvironment
         public string ConnectionString => GetRecordedOptionalVariable("CONNECTION_STRING");
         public string StorageAccountId => GetRecordedOptionalVariable("STORAGE_ID");
         public string StorageAccountConnectionString => GetRecordedOptionalVariable("STORAGE_CONNECTION_STRING");
-        public string MetricsLocation => GetRecordedVariable("METRICS_LOCATION");
```
Comment (JoshLove-msft) [inline sdk/monitor/Azure.Monitor.Query/tests/MonitorQueryTestEnvironment.cs:31]: Does this need to be deleted from the bicep?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.Query/tests/MonitorQueryTestEnvironment.cs (hunk 5 above)
Comment (nisha-bhatia) [reply sdk/monitor/Azure.Monitor.Query/tests/MonitorQueryTestEnvironment.cs:31]: I think it is deleted :)

**Pull request merged**